pkg/game/position/position.go

```diff
@@ -0,0 +1,70 @@
+package position
+
+import (
+	"fmt"
+
+	sideMod "github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+)
+
+type Position struct {
+	// int8 would be fine for base game (72 tiles) but let's be a bit more generous
+	x int16
+	y int16
+}
+
+func New(x int16, y int16) Position {
+	return Position{x, y}
+}
+
+func (pos Position) X() int16 {
+	return pos.x
+}
+
+func (pos Position) Y() int16 {
+	return pos.y
+}
+
+func (pos Position) Add(other Position) Position {
+	return New(pos.x+other.x, pos.y+other.y)
+}
+
+/*
+Returns position neighbouring (0,0) from the given side.
+Valid sides are either a single half-edge side (e.g. side.TopLeftEdge) or a single primary side (e.g. side.Right)
+Examples:
+position.FromSide(side.Right) ---> (1,0)
+position.FromSide(side.BottomLeftEdge) == position.FromSide(side.BottomRightEdge) ---> (0, -1)
+*/
+func FromSide(side sideMod.Side) Position {
+	primarySides := 0
+	for _, checkedSide := range []sideMod.Side{sideMod.Top, sideMod.Right, sideMod.Left, sideMod.Bottom} {
+		if side&checkedSide != 0 {
+			primarySides++
+		}
+	}
+
+	if primarySides == 0 {
+		return New(0, 0)
+	} else if primarySides == 1 {
+		switch {
+		case side&sideMod.Top != 0:
+			return New(0, 1)
+		case side&sideMod.Right != 0:
+			return New(1, 0)
+		case side&sideMod.Left != 0:
+			return New(-1, 0)
+		case side&sideMod.Bottom != 0:
+			return New(0, -1)
+		}
+	}
+	panic(fmt.Sprintf("position.FromSide called with more than one primary side. 'side' = %08b", side))
+}
```

Comment:
```suggestion
func FromSide(side sideMod.Side) Position {
	switch side {
	case sideMod.Top:
		return New(0, 1)
	case sideMod.Right:
		return New(1, 0)
	case sideMod.Left:
		return New(-1, 0)
	case sideMod.Bottom:
		return New(0, -1)
	default:
		panic(fmt.Sprintf("position.FromSide called with more than one primary side. 'side' = %08b", side))
	}
}
```

Reply:
This won't work as intended, as for example (0, 1) should be returned not only when side=Top, but also when it's equal to TopRightCorner and TopLeftCorner. That's why there is & and checking if it's != 0

(see the documentation comment for more details)

Reply:
Ok, then only tests that feature invalid cases (e.g. combination of sideMod.Left and sideMod.BottomRightEdge) are missing.

Reply:
They are not missing.
In the `TestPositionFromSidePanic` function (in position_test.go) the following invalid cases are tested:
```
side.Top | side.Left,
side.RightBottomEdge | side.BottomRightEdge,
side.Top | side.Right | side.Bottom | side.Left,
side.Top | side.BottomLeftEdge,
```